From 2add63ff0861c0af4f51046f6ffdf79e2c5e270d Mon Sep 17 00:00:00 2001
From: Jerome Neanne <jneanne@baylibre.com>
Date: Fri, 8 Jul 2022 19:14:36 +0200
Subject: [PATCH 10/18] regulator: dt-bindings: Add interrupts support to
 TPS65219 PMIC bindings

Add interrupt properties in PMIC TPS65219 bindings

Signed-off-by: Jerome Neanne <jneanne@baylibre.com>
Signed-off-by: Nishanth Menon <nm@ti.com>
---
 .../bindings/regulator/ti,tps65219.yaml       | 23 +++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/Documentation/devicetree/bindings/regulator/ti,tps65219.yaml b/Documentation/devicetree/bindings/regulator/ti,tps65219.yaml
index a4717ff4e95b..837420712ee6 100644
--- a/Documentation/devicetree/bindings/regulator/ti,tps65219.yaml
+++ b/Documentation/devicetree/bindings/regulator/ti,tps65219.yaml
@@ -20,6 +20,17 @@ properties:
   reg:
     maxItems: 1
 
+  interrupts:
+    description: Short-circuit, over-current, under-voltage for regulators, PB interrupts.
+    maxItems: 1
+
+  interrupt-controller: true
+
+  '#interrupt-cells':
+    description: Specifies the PIN numbers and Flags, as defined in
+      include/dt-bindings/interrupt-controller/irq.h
+    const: 2
+
 patternProperties:
   "^buck[1-3]-supply$":
     description: Input supply phandle of one regulator.
@@ -66,12 +77,17 @@ patternProperties:
 required:
   - compatible
   - reg
+  - interrupts
+  - interrupts
+  - interrupt-controller
+  - '#interrupt-cells'
   - regulators
 
 additionalProperties: false
 
 examples:
   - |
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
     i2c {
         #address-cells = <1>;
         #size-cells = <0>;
@@ -87,6 +103,13 @@ examples:
             ldo3-supply = <&vcc_3v3_sys>;
             ldo4-supply = <&vcc_3v3_sys>;
 
+            pinctrl-0 = <&pmic_irq_pins_default>;
+
+            interrupt-parent = <&gic500>;
+            interrupts = <GIC_SPI 224 IRQ_TYPE_LEVEL_HIGH>;
+            interrupt-controller;
+            #interrupt-cells = <2>;
+
             regulators {
                 buck1_reg: buck1 {
                     regulator-name = "VDD_CORE";
-- 
2.30.2

